### PR TITLE
[W5.11][T09-A2] Chen Xing

### PR DESCRIPTION
**Pull request by chenxing1992**
1)Display automatically all person out after add
2) Able to Sort the Person list in alphabetical order

### CODE REVIEW

**Comment by kengsengg** [comment]
Good attempt of implementing the enhancements with multiple small commits that are easy to understand. However, you may want to use different branches for different learning outcomes, so that your pull requests will be less messy. Also, do take note of the indentation of your codes.

**Review by kengsengg — COMMENTED** [review]
Try to ensure consistent indentation of the code and the extra blank lines can be removed to make the code more neat and concise

File: doc/UserGuide.md [hunk 1]
```diff
@@ -48,6 +48,9 @@ Format: `add NAME [p]p/PHONE_NUMBER [p]e/EMAIL [p]a/ADDRESS [t/TAG]...`
 Examples: 
 * `add John Doe p/98765432 e/johnd@gmail.com a/John street, block 123, #01-01`
 * `add Betsy Crowe pp/1234567 e/betsycrowe@gmail.com pa/Newgate Prison t/criminal t/friend`
+### Sort all persons : `sort`
+Sorting all person from the address book in alphabetical ordering.
+Format: `sorting`
```
Comment (dalessr) [inline doc/UserGuide.md:53]: "sorting" command name is not consistent with all the other command names like "add" "list" "find". It is btter to just use "sort" as command word.

File: src/seedu/addressbook/Main.java [hunk 2]
```diff
@@ -4,16 +4,21 @@
 import java.util.List;
 import java.util.Optional;
 
+
+import seedu.addressbook.commands.AddCommand;
 import seedu.addressbook.commands.Command;
 import seedu.addressbook.commands.CommandResult;
 import seedu.addressbook.commands.ExitCommand;
+
+import seedu.addressbook.commands.*;
```
Comment (dalessr) [inline src/seedu/addressbook/Main.java:13]: This is a coding style violation, you need to specify the exact class/interface imported.

File: src/seedu/addressbook/Main.java [hunk 3]
```diff
@@ -4,16 +4,21 @@
...
 import seedu.addressbook.data.AddressBook;
 import seedu.addressbook.data.person.ReadOnlyPerson;
 import seedu.addressbook.parser.Parser;
 import seedu.addressbook.storage.StorageFile;
 import seedu.addressbook.storage.StorageFile.InvalidStorageFilePathException;
 import seedu.addressbook.storage.StorageFile.StorageOperationException;
 import seedu.addressbook.ui.TextUi;
+import seedu.addressbook.ui.Formatter;
```
Comment (dalessr) [inline src/seedu/addressbook/Main.java:21]: The import classes should also follow the alphabetical order (which means TextUi should be after Formatter). For AB Lvl 3 & 4 there will be checkstyles in the travis test.

File: src/seedu/addressbook/Main.java [hunk 4]
```diff
@@ -48,14 +62,18 @@ public void run(String[] launchArgs) {
...
-     *
      */
     private void start(String[] launchArgs) {
         try {
             this.ui = new TextUi();
+            this.formatter = new Formatter();
             this.storage = initializeStorage(launchArgs);
             this.addressBook = storage.load();
             ui.showWelcomeMessage(VERSION, storage.getPath());
+            command = new Parser().parseCommand(ui.getUserCommand());
```
Comment (dalessr) [inline src/seedu/addressbook/Main.java:73]: Use blank lines sparingly. It's good to have blank lines to separate logical code blocks, but too many blank lines will increase the height of the code, which increases the amount of scrolling.

File: src/seedu/addressbook/data/person/Address.java [hunk 5]
```diff
@@ -6,29 +6,49 @@
  * Represents a Person's address in the address book.
  * Guarantees: immutable; is valid as declared in {@link #isValidAddress(String)}
  */
-public class Address {
+public class Address extends Contact {
+
+    public static final String EXAMPLE = "a/123, Clementi Ave 3, #12-34, 231534";
+    // public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
```
Comment (dalessr) [inline src/seedu/addressbook/data/person/Address.java:12]: Don't leave commented out code. Just go ahead and delete unused code. Now that we are using a revision control system, we can recover any deleted code later.

File: src/seedu/addressbook/data/person/Person.java [hunk 6]
```diff
@@ -14,8 +14,11 @@
     private Phone phone;
     private Email email;
     private Address address;
+    private int seqNumber;
+    private static int nextSeqNumber = 1;
```
Comment (dalessr) [inline src/seedu/addressbook/data/person/Person.java:18]: This is not the correct naming convention for a constant value (all upper cases with '_')

File: src/seedu/addressbook/data/person/ReadOnlyPerson.java [hunk 7]
```diff
@@ -37,11 +37,11 @@ default boolean isSamePerson(ReadOnlyPerson other) {
     default boolean hasSameData(ReadOnlyPerson other) {
         return other == this // short circuit if same object
                 || (other != null // this is first to avoid NPE below
-                    && other.getName().equals(this.getName()) // state checks here onwards
-                    && other.getPhone().equals(this.getPhone())
-                    && other.getEmail().equals(this.getEmail())
-                    && other.getAddress().equals(this.getAddress())
-                    && other.getTags().equals(this.getTags()));
+                && other.getName().equals(this.getName()) // state checks here onwards
+                               && other.getPhone().equals(this.getPhone())
```
Comment (dalessr) [inline src/seedu/addressbook/data/person/ReadOnlyPerson.java:41]: The clauses need to be align to each other.

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 8]
```diff
@@ -33,14 +33,16 @@ protected DuplicatePersonException() {
      * Signals that an operation targeting a specified person in the list would fail because
      * there is no such matching person in the list.
      */
-    public static class PersonNotFoundException extends Exception {}
+    public static class PersonNotFoundException extends Exception {
```
Comment (dalessr) [inline src/seedu/addressbook/data/person/UniquePersonList.java:36]: This formatting is actually not necessary. Just leave it as original.

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 9]
```diff
@@ -100,8 +103,8 @@ public boolean contains(ReadOnlyPerson toCheck) {
      * Adds a person to the list.
      *
      * @throws DuplicatePersonException if the person to add is a duplicate of an existing person in the list.
-     *    The @link{ReadOnlyPerson#isSamePerson} method is used for this comparison,
-     *    which defines a weaker notion of equality.
+     *                                  The @link{ReadOnlyPerson#isSamePerson} method is used for this comparison,
```
Comment (dalessr) [inline src/seedu/addressbook/data/person/UniquePersonList.java:106]: Eh why there are so many spaces here?

File: src/seedu/addressbook/parser/Parser.java [hunk 10]
```diff
@@ -75,33 +82,37 @@ public Command parseCommand(String userInput) {
 
         switch (commandWord) {
 
-        case AddCommand.COMMAND_WORD:
-            return prepareAdd(arguments);
+            case AddCommand.COMMAND_WORD:
```
Comment (dalessr) [inline src/seedu/addressbook/parser/Parser.java:85]: the "case" should be align with "switch". Note that sometimes the coding style in Intellij is different from what we expect.

**Review by dalessr — COMMENTED** [review]
Good job on covering all aspects of the new command @chenxing1992  👍Please note that you need to create to PRs for the two distinct features :)

Some comments added. Please close the PR after reading comments.

**Pull request closed without merging**